activestorage/lib/active_storage/attached/macros.rb

```diff
@@ -83,7 +83,21 @@ def #{name}=(attachables)
         end
       CODE
 
-      has_many :"#{name}_attachments", -> { where(name: name) }, as: :record, class_name: "ActiveStorage::Attachment", inverse_of: :record
+      has_many :"#{name}_attachments", -> { where(name: name) }, as: :record, class_name: "ActiveStorage::Attachment", inverse_of: :record do
+        def purge
+          if any?
+            each(&:purge)
+            reset
+          end
+        end
+
+        def purge_later
+          if any?
+            each(&:purge_later)
+            reset
+          end
+        end
+      end
```

Comment:
Could we define these as class methods on `ActiveStorage::Attachment` instead? Then users would also be free to do `ActiveStorage::Attachment.where(**).purge` from a console session, for instance.

Reply:
I'm not sure I know how to do this.

Reply:
Try opening up the class with this in `ActiveStorage::Attachment`.

```ruby
class << self
  # The class methods go here.
end
```